content/posts/2018-04-19--how-to-implement-redux-in-react/index.md

```diff
@@ -0,0 +1,869 @@
+---
+title: How to implement Redux in React
+subTitle: What is Redux and how we can combine it with our application?
+category: "Redux"
+cover: redux.png
+postAuthor: Michał Rożenek
+---
+
+# What is Redux?
+
+Everything you change in your application, included data and UI changes is contained in an object called `state`, so
+basicaly Redux is a store which contains application state.
+
+## State 
+
+`State` is read only.
+When you want to modify the state,
+you need to `dispatch` an `action`.
+
+## Action
+
+`Action` is a plain javascript object, described in the simple way, includes an informations about changes in the application.
+Action minimum requirement is a `TYPE` property. Actions could receive some arguments. Actions in redux have to be pure!
+
+## Reducer
+
+The state mutations in your app need to be pure functions.
+Reducer receive initial state, previos state and action.
+Reducer don't modify the previous state, just return a new object.
+
+## Store
+
+`Store` has a current application `state` object and let you `dispatch actions`.
+During creating a store, you need to provide a `reducer` as a parameter to let the store know what and how you want to update.
+
+### Store methods:
+
+`getState` - return current state
+
+`dispatch` - let you dispatch actions
+
+`subscribe` - let you subscribe on store changes and register a callback
+
+Store shouldn't be mutated!
+You should always return a new object.
+To avoild an Object mutations, you can use those methods:
+
+- spread operator:
+```js
+return {
```

Comment:
Something is wrong with indentation on preview here

Reply:
I don't know why :)
I tried to fix it... but it don't want to listen to me :smile: